src/main/java/com/conveyal/datatools/manager/jobs/validation/SharedStopsValidator.java

```diff
@@ -58,11 +56,40 @@ public void validate() {
 
         CsvReader configReader = CsvReader.parse(config);
 
-        // TODO: pull indicies from the CSV header
-        final int STOP_GROUP_ID_INDEX = 0;
-        final int STOP_ID_INDEX = 2;
-        final int IS_PRIMARY_INDEX = 3;
-        final int FEED_ID_INDEX = 1;
+        int STOP_GROUP_ID_INDEX = -1;
+        int STOP_ID_INDEX = -1;
+        int IS_PRIMARY_INDEX = -1;
+        int FEED_ID_INDEX = -1;
+
+        try {
+            configReader.setHeaders(new String[]{"stop_group_id", "feed_id", "stop_id", "is_primary"});
```

Comment:
My understanding is that you are setting the headers (and presumably the order) and then checking the index of each. I think the index will always be the same for each header as defined here.

Also, if the order in `project.sharedStopsConfig` doesn't match this, the headers will be assigned to the wrong data column. To fix (assuming my understanding of this is correct) is to remove this line and therefore use the headers as they are defined (rightly or wrongly) in `project.sharedStopsConfig`. A unit test or two around this would be really useful.

Reply:
Do we have examples of unit testing other validators? I've manually created some tests to fix the behavior issues you're talking about but I can't find a good way to split things to reasonably test them. Either way the issues you've mentioned have been fixed. Using `split`.

Reply:
@miles-grant-ibigroup Look to move the contents of `validate()` method into a static method or two so these can be called from a unit test to avoid having to to go through the whole feed/validation process. 

See https://github.com/ibi-group/gtfs-lib/blob/dev-flex/src/main/java/com/conveyal/gtfs/validator/FlexValidator.java and https://github.com/ibi-group/gtfs-lib/blob/dev-flex/src/test/java/com/conveyal/gtfs/validator/FlexValidatorTest.java.

Reply:
I've given this an attempt. Let me know what you think

Reply:
@miles-grant-ibigroup I hope you don't mind, but I have created a PR of off this PR: https://github.com/ibi-group/datatools-server/pull/585 and made a few changes. Let me know if you have any questions. If not, you can merge that back into this PR.